src/fsharp/CompileOps.fs

```diff
@@ -1700,6 +1700,7 @@ let DefaultReferencesForScriptsAndOutOfProjectSources(assumeDotNetFramework) =
           yield "System.Numerics" 
       else
           yield Path.Combine(Path.GetDirectoryName(typeof<System.Object>.Assembly.Location),"mscorlib.dll"); // mscorlib
+          yield typeof<System.Object>.Assembly.Location; // System.Private.CoreLib
```

Comment:
We need to get to the bottom of why this is needed. AFAIK it should never be necessary to reference System.Private.CoreLib or any implementation assemblies which in turn refer to this.

I'll admit I find the facade/implementation assembly split bewildering in practice and the details always slip away.  Do you know what's causing the need for this reference?  Thanks